tests/core/test_domain.py

```diff
@@ -616,17 +631,18 @@ def test_clean_domain():
             "utter_default": [{"text": "default message"}],
         },
         "actions": ["utter_default", "utter_goodbye", "utter_greet"],
+        "session_config": {
```

Comment:
Previously, this test only checked if the (hashes of) the domains resulting from the two dictionaries are the same. I think that doesn't achieve what it should. Say `actual` is the output of the cleaning function and `expected` is a properly cleaned domain. Even if the cleaning function did nothing, `assert hash(Domain.from_dict(actual))==hash(Domain.from_dict(expected))` would still pass.
So we have to directly compare the dictionaries to really test if the cleaning function properly cleans the domain.

Since a recent change in rasa, domains have a `session_config` property. Comparing the domains still worked because a default value is added when calling `Domain.from_dict(some_dict)`. For comparing the dicts directly, I had to manually add this property to the `expected` dict.

Reply:
I think this test just tries to ensure that the format in `default_unfeaturized_entities.yml` is already properly cleaned, so that the `as_dict()` versions of `expected` and `default_unfeaturized_entities.yml` are the same (done by `domain.__hash__()`). both are fine, your test covers that implicitly 👍